crates/biome_js_analyze/src/lint/style/use_consistent_array_type.rs

```diff
@@ -105,23 +105,30 @@ impl Rule for UseConsistentArrayType {
                 if options.syntax == ConsistentArrayType::Shorthand {
                     return None;
                 }
-                match get_array_kind_by_any_type(query) {
-                    Some(array_kind) => transform_array_type(query.to_owned(), array_kind),
-                    _ => None,
-                }
+                let array_kind = get_array_kind_by_any_type(query)?;
+                transform_array_type(query.to_owned(), array_kind)
             }
             AnyTsType::TsReferenceType(ty) => {
                 if options.syntax == ConsistentArrayType::Generic {
                     return None;
                 }
 
-                match get_array_kind_by_reference(ty) {
-                    Some(array_kind) => {
-                        let type_arguments = &ty.type_arguments()?;
-                        convert_to_array_type(type_arguments, array_kind)
-                    }
-                    _ => None,
+                // Ignore `Array` in an `extends` clause.
+                let parent = ty
+                    .syntax()
+                    .ancestors()
+                    // Skip `TsReferenceType` and `TsTypeList`
+                    .skip(2)
```

Comment:
It makes sense to use `skip(1)` to skip self, but I think assuming that the CST structure won't change in the future isn't reliable. 

If tomorrow the structure of node changes, the rule breaks

Reply:
Good point

Reply:
This will be caught by a failing test? Are you suggesting to explicitly ensure that the parent is `TsTypeList`?

Reply:
Thanks for the review. I actually find a code I forgot to handle: `implements Array<X>`.